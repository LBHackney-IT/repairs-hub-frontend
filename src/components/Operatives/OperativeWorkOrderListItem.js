import PropTypes from 'prop-types'

const OperativeWorkOrderListItem = ({ operativeWorkOrders }) => (
  <>
    <ol className="lbh-list">
      {operativeWorkOrders.map((operativeWorkOrder, index) => (
        <li key={index} className="green-info-box" data-note-id={index}>
          <div className="appointment-details">
            <p className="lbh-body lbh-!-font-weight-bold text-white">{`${operativeWorkOrder.appointment.start}-${operativeWorkOrder.appointment.end}`}</p>

            <p className="lbh-body lbh-!-font-weight-bold text-white">
              {operativeWorkOrder.priority}
            </p>
          </div>
          <p className="middle-right">
            <i className="arrow right"></i>
          </p>
        </li>
      ))}
    </ol>
  </>
)

OperativeWorkOrderListItem.propTypes = {
  operativeWorkOrders: PropTypes.arrayOf(
    PropTypes.shape({
      reference: PropTypes.number,
      dateRaised: PropTypes.string,
      lastUpdated: PropTypes.instanceOf(Date),
      priority: PropTypes.string,
      property: PropTypes.string,
      owner: PropTypes.string,
      description: PropTypes.string,
      propertyReference: PropTypes.string,
      tradeCode: PropTypes.string,
      tradeDescription: PropTypes.string,
      status: PropTypes.string,
    })
  ).isRequired,
}

export default OperativeWorkOrderListItem
