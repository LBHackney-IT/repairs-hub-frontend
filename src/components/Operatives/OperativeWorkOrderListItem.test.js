import { render } from '@testing-library/react'
import OperativeWorkOrderListItem from './OperativeWorkOrderListItem'

describe('OperativeWorkOrderListItem component', () => {
  const props = {
    operativeWorkOrder: {
      reference: 10000621,
      dateRaised: '2021-06-11T13:49:15.878796Z',
      lastUpdated: null,
      priority: '5 [N] NORMAL',
      property: '17 Pitcairn House  St Thomass Square',
      propertyPostCode: 'L53 GS',
      owner: 'Herts Heritage Ltd',
      description: 'r',
      propertyReference: '00023405',
      tradeCode: 'PL',
      tradeDescription: 'Plumbing - PL',
      status: 'In Progress',
      appointment: {
        date: '2021-09-03',
        description: 'AM Slot',
        start: '08:00',
        end: '13:00',
        reason: null,
        note: null,
      },
    },
  }

  it('should render operativeWorkOrderListItem', () => {
    const { asFragment } = render(
      <OperativeWorkOrderListItem
        operativeWorkOrder={props.operativeWorkOrder}
        index={0}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
