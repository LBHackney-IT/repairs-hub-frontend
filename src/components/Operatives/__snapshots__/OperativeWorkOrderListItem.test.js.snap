// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperativeWorkOrderListItem component should render operativeWorkOrderListItem 1`] = `
<DocumentFragment>
  <li
    class="govuk-!-margin-top-3 operative-work-order-list-item operative-work-order-list-item--active"
    data-id="0"
  >
    <div
      class="appointment-details"
    >
      <div>
        <h3
          class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-0 govuk-!-display-inline"
        >
          08:00 â€“ 13:00
        </h3>
      </div>
      <p
        class="lbh-body lbh-!-font-weight-bold govuk-!-margin-0 govuk-!-margin-bottom-2 capitalize"
      >
        normal
      </p>
      <h3
        class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-0 govuk-!-display-inline"
      >
        WO 10000621
      </h3>
      <p
        class="lbh-body govuk-!-margin-0"
      >
        17 Pitcairn House  St Thomass Square
      </p>
      <p
        class="lbh-body govuk-!-margin-0 govuk-!-margin-bottom-8"
      >
        L53 GS
      </p>
      <p
        class="lbh-body govuk-!-margin-0 truncate-description truncate-line-3"
      >
        Lorem ipsum dolor sit amet, consectetur efficitur.
      </p>
    </div>
    <div
      class="govuk-!-margin-0"
    >
      <span
        class="arrow right"
      />
    </div>
  </li>
</DocumentFragment>
`;
